test/transforms.jl

```diff
@@ -694,6 +694,84 @@
     @test n1 == n2
   end
 
+  @testset "StdNames" begin
+    names = ["apple banana", "apple_banana", "apple_Banana"]
+    for name in names
+      @test TableTransforms._camel(name) == "AppleBanana"
+      @test TableTransforms._snake(name) == "apple_banana"
+      @test TableTransforms._upper(name) == "APPLEBANANA"
+    end
+
+    names = ["a", "A", "_a", "_A", "a ", "A "]
+    for name in names
+      @test TableTransforms._camel(name) == "A"
+      @test TableTransforms._snake(name) == "a"
+      @test TableTransforms._upper(name) == "A"
+    end
+
+    # special characters
+    
+    name = "a&B"
+    @test TableTransforms._camel(name) == "A&B"
+    @test TableTransforms._snake(name) == "a&b"
+    @test TableTransforms._upper(name) == "A&B"
+
+    name = "apple-tree"
+    @test TableTransforms._camel(name) == "Apple-tree"
+    @test TableTransforms._snake(name) == "apple-tree"
+    @test TableTransforms._upper(name) == "APPLE-TREE"
+    
+    name = "apple#"
+    @test TableTransforms._camel(name) == "Apple#"
+    @test TableTransforms._snake(name) == "apple#"
+    @test TableTransforms._upper(name) == "APPLE#"
```

Comment:
I think we should erase all special characters (except the `delim` characters) in a pre-processing step inside the `apply` function. @eliascarv do you also think this is a good idea?

I think we should set `delim = [' ', '\t', '-', '_']` as our possible delimiters. All other characters could be erased.

Reply:
I would also suggest adding `&` to the list of delimiters. That being said, I don't think we can make this transform work well with complex column names, anyway. Maybe we can later add more specific transforms that deal with column names (such as `Replace`, but for column names). @juliohm @eliascarv